search/reverse_geocoder.cpp

```diff
@@ -275,8 +275,6 @@ void ReverseGeocoder::GetNearbyBuildings(m2::PointD const & center, double radiu
   auto const addBuilding = [&](FeatureType & ft)
   {
     std::string const & hn = GetHouseNumber(ft);
-    if (hn.empty())
```

Comment:
Probably it should be changed to "return; if there is no address".

Your current change makes the reverse geocoder to return buildings with no address at all, i.e. useless.

That's likely the cause of the failing test too. The rev geocoder stops searching after reaching `kMaxNumTriesToApproxAddress` - perhaps address-less buildings take all those slots and the best matching (closest) building never gets in there.